op-acceptance-tests/tests/jovian/min_base_fee_test.go

```diff
@@ -0,0 +1,197 @@
+	prevBlockNum := info.NumberU64()
+	for range 5 {
+		n := mbf.getBlock()
+		mbf.require.True(n.NumberU64() > prevBlockNum, "block number should increase")
+
+		prevBlockNum = n.NumberU64()
+		feeCmp := n.BaseFee().Cmp(minBase)
+		if !shouldEnforce {
+			mbf.require.True(feeCmp > 0, "expected base fee to be higher than the minBaseFee")
+		} else {
+			mbf.require.True(feeCmp == 0, "expected base fee to be at least minBaseFee")
+		}
+		mbf.t.Logf("base fee %s, minBase %s %d", n.BaseFee(), minBase)
+	}
+}
+
+// waitForMinBaseFee waits until the L2 latest payload extra-data encodes the expected min base fee.
+func (mbf *minBaseFee) waitForMinBaseFee(expected uint64) {
+	client := mbf.l2EL.Escape().L2EthClient()
+	ext, ok := client.(apis.L2EthExtendedClient)
+	mbf.require.True(ok, "L2 client does not support extended payload API")
+
+	expectedExtraData := eth.BytesMax32(eip1559.EncodeJovianExtraData(250, 6, expected))
+
+	var actualPayload eth.BytesMax32
+	mbf.require.Eventually(func() bool {
+		payload, err := ext.PayloadByLabel(mbf.ctx, "latest")
+		if err != nil {
+			return false
+		}
+		if len(payload.ExecutionPayload.ExtraData) != 17 {
+			return false
+		}
+
+		got := binary.BigEndian.Uint64(payload.ExecutionPayload.ExtraData[9:])
+		actualPayload = payload.ExecutionPayload.ExtraData
+		return got == expected
+	}, 2*time.Minute, 5*time.Second, "L2 min base fee did not sync within timeout")
+
+	mbf.require.Equal(expectedExtraData, actualPayload, "extradata doesnt match")
+}
+
+func (mbf *minBaseFee) getBlock() eth.BlockInfo {
+	_ = mbf.l2EL.WaitForBlock()
+	el := mbf.l2EL.Escape().EthClient()
+	info, err := el.InfoByLabel(mbf.ctx, "latest")
+	mbf.require.NoError(err)
+	return info
+}
+
+// TestMinBaseFee verifies configurable minimum base fee using devstack presets.
+func TestMinBaseFee(gt *testing.T) {
+	t := devtest.SerialT(gt)
+	sys := presets.NewMinimal(t)
+	require := t.Require()
+
+	err := dsl.RequiresL2Fork(t.Ctx(), sys, 0, rollup.Jovian)
+	require.NoError(err, "Jovian fork must be active for this test")
+
+	fundAmount := eth.OneTenthEther
+	alice := sys.FunderL2.NewFundedEOA(fundAmount)
+
+	alice.WaitForBalance(fundAmount)
+	bob := sys.Wallet.NewEOA(sys.L2EL)
+
+	minBaseFee := newMinBaseFee(t, sys.L2Chain, sys.L1EL, sys.L2EL)
+	minBaseFee.checkCompatibility()
+
+	systemOwner := minBaseFee.getSystemOwner()
+	sys.FunderL1.FundAtLeast(systemOwner, eth.OneTenthEther)
+
+	testCases := []struct {
+		name          string
+		minBaseFee    uint64
+		shouldEnforce bool
+	}{
+		// The min base fee is set too low so when there's activity, we enforce the
+		// calculated base fee over the min base fee.
+		{"MinBaseFeeNotEnforced", 0, false},
+		// The min base fee is enforced since the calculated base fee is below the min base fee.
+		{"MinBaseFeeEnforced", 1_000_000_000, true},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t devtest.T) {
+			minBaseFee.setMinBaseFee(tc.minBaseFee)
+			minBaseFee.waitForMinBaseFee(tc.minBaseFee)
```

Comment:
Nit: something like this might be more readable?

```suggestion
			minBaseFee.setMinBaseFeeViaSytemConfigOnL1(tc.minBaseFee)
			minBaseFee.waitForMinBaseFeeConfigChangeOnL2(tc.minBaseFee)
```